v-next/hardhat/src/internal/cli/telemetry/sentry/send-to-sentry.ts

```diff
@@ -0,0 +1,80 @@
+import type { Event } from "@sentry/node";
+
+import { writeUtf8File } from "@ignored/hardhat-vnext-utils/fs";
+import { captureEvent, captureMessage, init } from "@sentry/node";
+
+import { Anonymizer } from "./anonymizer.js";
+import { SENTRY_DSN } from "./reporter.js";
+
+export async function main(): Promise<void> {
+  try {
+    init({
+      dsn: SENTRY_DSN,
+    });
+  } catch (error) {
+    process.exit(1);
+  }
+
+  const serializedEvent = process.argv[2];
+  const configPath = process.argv[3];
+
+  if (serializedEvent === undefined) {
+    await sendMsgToSentry(
+      "There was an error parsing an event: 'process.argv[2]' argument is not set",
+    );
+    return;
+  }
+
+  let event: any;
+  try {
+    event = JSON.parse(serializedEvent);
+  } catch {
+    await sendMsgToSentry(
+      "There was an error parsing an event: 'process.argv[2]' doesn't have a valid JSON",
+    );
+    return;
+  }
+
+  try {
+    const anonymizer = new Anonymizer(configPath);
+    const anonymizedEvent = await anonymizer.anonymize(event);
+
+    if (anonymizedEvent.isRight()) {
+      if (anonymizer.raisedByHardhat(anonymizedEvent.value)) {
```

Comment:
This could be in the reporter, and I'm not convinced that this needs to be part of the anonymizer

Reply:
As discussed in the messages, we will keep it here because the errors are stringified by Sentry in an easy-to-analyze format. Hence, it's easy to check the file paths, etc.